src/extension.test.ts

```diff
@@ -98,6 +99,54 @@ describe("ExtensionContext", () => {
   }
 });
 
+describe("Refreshable views tests", () => {
+  before(async () => {
+    await getTestExtensionContext();
+  });
+
+  it("getRefreshableViewProviders returns unique view providers + name pairs", () => {
+    const seenNames = new Set();
+    const seenViewProviderConstructorNames = new Set();
+
+    for (const [viewProvider, name] of getRefreshableViewProviders()) {
+      assert.ok(
+        !seenNames.has(name),
+        `Duplicate name "${name}" found in refreshable view providers`,
+      );
+      seenNames.add(name);
+
+      assert.ok(
+        !seenViewProviderConstructorNames.has(viewProvider.constructor.name),
+        `Duplicate view provider constructor "${viewProvider.constructor.name}" found`,
+      );
+      seenViewProviderConstructorNames.add(viewProvider.constructor.name);
+    }
+  });
+
+  it("_activateExtension should have registered refresh commands for expected view providers", async () => {
+    const viewProviderNameFragments = ["resources", "topics", "schemas", "statements"];
```

Comment:
Minor: could get annoying to update this if/when we add more views since it'll be another place to update; would it be easier to use the name values from getRefreshableViewProviders() here instead?

Reply:
But how else to prove that the exact views we literally expect to be refreshable are actually refreshable? Only ever depending on the tested function's return results will completely miss that "oh we forgot to wire up refreshability into new view Foo". Something in test suite side needs to be grounded in facts, not the data-generating-function's return results.